sonic-chassisd/scripts/chassisd

```diff
@@ -603,6 +808,52 @@ class ConfigManagerTask(ProcessTaskBase):
 
             self.config_updater.module_config_update(key, admin_state)
 
+
+#
+# SmartSwitch Config Manager task ========================================================
+#
+
+
+class SmartSwitchConfigManagerTask(ProcessTaskBase):
+    def __init__(self):
+        ProcessTaskBase.__init__(self)
+
+        # TODO: Refactor to eliminate the need for this Logger instance
+        self.logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+    def task_worker(self):
+        self.config_updater = SmartSwitchModuleConfigUpdater(SYSLOG_IDENTIFIER, platform_chassis)
+        config_db = daemon_base.db_connect("CONFIG_DB")
+
+        # Subscribe to CHASSIS_MODULE table notifications in the Config DB
+        sel = swsscommon.Select()
+        sst = swsscommon.SubscriberStateTable(config_db, CHASSIS_CFG_TABLE)
+        sel.addSelectable(sst)
+
+        # Listen indefinitely for changes to the CFG_CHASSIS_MODULE_TABLE table in the Config DB
+        while True:
+            # Use timeout to prevent ignoring the signals we want to handle
+            # in signal_handler() (e.g. SIGTERM for graceful shutdown)
+            (state, c) = sel.select(SELECT_TIMEOUT)
+
+            if state == swsscommon.Select.TIMEOUT:
+                # Do not flood log when select times out
+                continue
+            if state != swsscommon.Select.OBJECT:
+                self.logger.log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+                continue
+
+            (key, op, fvp) = sst.pop()
+
+            if op == 'SET':
+                admin_state = MODULE_ADMIN_DOWN
+            elif op == 'DEL':
```

Comment:
@rameshraghupathy on DEL of the admin state why do module ADMIN UP?

Reply:
> @rameshraghupathy on DEL of the admin state why do module ADMIN UP?

@prgeor The original behavior is: The default is no config - meaning admin up   when the user wants to shutdown a module adds a config "admins_status: down".  So, the SET operation has been used for "admins_status: down" and DEL has been used for setting the module up.   If you want we can change this for Smartswitch.  Please let me know.

Reply:
@rameshraghupathy didn't we agree to keep the default behavior on missing of the config DB admin status to have dark mode enabled as default? @vvolam is that not captured in the HLD?